GitWorkflow/GitWorkflow/Player.cpp

```diff
@@ -62,23 +62,49 @@ void Player::updateMovement(const float dt)
 	}
 	else if (m_velocity.x > -m_speed)
 		accelerate({ -m_acceleration, 0.0f }, dt);
-	else if(m_velocity.x < -m_speed && !m_sprinting && isColliding()) 
-		accelerate({  m_acceleration, 0.0f }, dt);
+	else if (m_velocity.x < -m_speed && !m_sprinting && isColliding())
+		accelerate({ m_acceleration, 0.0f }, dt);
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) m_velocity.x -= m_speed;
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) m_velocity.x += m_speed;
+	if (!m_movingRight)
+	{
+		if (m_velocity.x > 0.0f)
+			accelerate({ isColliding() ? -m_acceleration : -m_acceleration * m_airControl, 0.0f }, dt);
+	}
+	else if (m_velocity.x < m_speed)
+		accelerate({ m_acceleration, 0.0f }, dt);
+	else if (m_velocity.x > m_speed && !m_sprinting && isColliding())
+		accelerate({ -m_acceleration, 0.0f }, dt);
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Space) && !m_hasJumped)
```

Comment:
In Player::accelerate(), I believe the two if-statements can be combined into one, but I haven't confirmed it yet.